systemtests/src/main/java/io/enmasse/systemtest/ability/TestWatcher.java

```diff
@@ -22,45 +23,43 @@
-                        testMethod.getName());
-                Files.createDirectories(path);
-                List<Pod> pods = kube.listPods();
-                for (Pod p : pods) {
-                    try {
-                        List<Container> containers = kube.getContainersFromPod(p.getMetadata().getName());
-                        for (Container c : containers) {
-                            File filePath = new File(path.toString(), String.format("%s_%s.log", p.getMetadata().getName(), c.getName()));
-                            try {
-                                Files.write(Paths.get(filePath.toString()), kube.getLog(p.getMetadata().getName(), c.getName()).getBytes());
-                            } catch (IOException e) {
-                                log.warn("Cannot write file {}", filePath.getName());
-                            }
+        try {
+            log.warn("Test failed: Saving pod logs and info...");
+            Kubernetes kube = Kubernetes.getInstance();
+            Path path = Paths.get(
+                    Environment.getInstance().testLogDir(),
+                    "failed_test_logs",
+                    testClass.getName(),
+                    testMethod.getName());
+            Files.createDirectories(path);
+            List<Pod> pods = kube.listPods();
+            for (Pod p : pods) {
+                try {
+                    List<Container> containers = kube.getContainersFromPod(p.getMetadata().getName());
+                    for (Container c : containers) {
+                        File filePath = new File(path.toString(), String.format("%s_%s.log", p.getMetadata().getName(), c.getName()));
+                        try {
+                            Files.write(Paths.get(filePath.toString()), kube.getLog(p.getMetadata().getName(), c.getName()).getBytes());
+                        } catch (IOException e) {
+                            log.warn("Cannot write file {}", filePath.getName());
                         }
-                    } catch (Exception ex) {
-                        log.warn("Cannot access logs from container: ", ex);
                     }
+                } catch (Exception ex) {
+                    log.warn("Cannot access logs from container: ", ex);
                 }
-                Files.write(Paths.get(path.toString(), "describe.txt"), KubeCMDClient.describePods(kube.getInfraNamespace()).getStdOut().getBytes());
-                Files.write(Paths.get(path.toString(), "events.txt"), KubeCMDClient.getEvents(kube.getInfraNamespace()).getStdOut().getBytes());
-                log.info("Pod logs and describe successfully stored into {}", path.toString());
-            } catch (Exception ex) {
-                log.warn("Cannot save pod logs and info: {}", ex.getMessage());
             }
+            Files.write(Paths.get(path.toString(), "describe.txt"), KubeCMDClient.describePods(kube.getInfraNamespace()).getStdOut().getBytes());
```

Comment:
Why not simply use `path.resolve("describe.txt")`?

Reply:
@ctron because I do not know it?

Reply:
@ctron fixed